core/src/main/scala/knobs/Configured.scala

```diff
@@ -41,11 +39,12 @@ object Configured {
     def apply(v: CfgValue) = f(v)
   }
 
-  implicit val configuredMonad: Monad[Configured] = new Monad[Configured] {
-    def point[A](a: => A) = new Configured[A] {
+  implicit val configuredMonad: Monad[Configured] = new StackSafeMonad[Configured] {
```

Comment:
Does anybody ever explicitly constrain to a `StackSafeMonad`, or does that exist solely for inheriting `tailRecM`?  I'm curious about the declared type of `configuredMonad`.

Reply:
Solely for inheriting `tailRecM` - I should probably implement this properly, I doubt `flatMap` itself is stack-safe.

Reply:
Actually I'm not sure how this would go - `Configured` is a glorified `Kleisli[IO, CfgValue, ?]` and there is https://github.com/typelevel/cats/issues/1733

Reply:
Something similar as circes' `Decoder` [`tailrecM`](https://github.com/circe/circe/blob/master/modules/core/shared/src/main/scala/io/circe/Decoder.scala#L868-L877) should be possible I think?